fe/fe-core/src/main/java/com/starrocks/scheduler/MvTaskRunProcessor.java

```diff
@@ -188,57 +193,64 @@ private void processPartitionWithPartitionTable(Database database, MaterializedV
-            for (Long refMvPartitionId : refMvPartitionIds) {
-                if (checkedMvPartitionIds.contains(refMvPartitionId)) {
+        Set<String> checkedMvPartitionNames = Sets.newHashSet();
+        for (String deletedPartitionName : deletedPartitionNames) {
+            // if base partition drop
+            Set<String> refMvPartitionNames = materializedView.getTableMvPartitionNameRefMap().get(deletedPartitionName);
+            for (String refMvPartitionName : refMvPartitionNames) {
+                if (checkedMvPartitionNames.contains(refMvPartitionNames)) {
                     continue;
                 }
-                checkedMvPartitionIds.add(refMvPartitionId);
-                Set<Long> refTablePartitionIds = partitionIdRefMap.get(refMvPartitionId);
-                if (deletedPartitionIds.containsAll(refTablePartitionIds)) {
-                    dropPartition(database, materializedView, refMvPartitionId);
+                checkedMvPartitionNames.add(refMvPartitionName);
+                Set<String> refTablePartitionNames = materializedView.getMvTablePartitionNameRefMap().get(refMvPartitionName);
+                if (deletedPartitionNames.containsAll(refTablePartitionNames)) {
+                    dropPartition(database, materializedView, refMvPartitionName);
                 } else {
-                    needRefreshPartitionIds.add(refMvPartitionId);
+                    needRefreshPartitionNames.add(refMvPartitionName);
                 }
             }
-            basePartitionVisionMap.remove(deletedPartitionId);
+            basePartitionVisionMap.remove(deletedPartitionName);
         }
-        // merge need refresh mv with comparing partition version
-        for (Long basePartitionId : basePartitionVisionMap.keySet()) {
-            Long oldVisibleVersion = basePartitionVisionMap.get(basePartitionId);
-            long visibleVersion = olapTable.getPartition(basePartitionId).getVisibleVersion();
-            if (visibleVersion > oldVisibleVersion) {
-                needRefreshPartitionIds.addAll(partitionIdRefMap.get(basePartitionId));
-                basePartitionVisionMap.put(basePartitionId, visibleVersion);
+        // merge need refresh mv with comparing other partitions
+        for (String basePartitionName : basePartitionVisionMap.keySet()) {
+            MaterializedView.BasePartitionInfo basePartitionInfo = basePartitionVisionMap.get(basePartitionName);
+            Partition partition = olapTable.getPartition(basePartitionName);
+            long visibleVersion = partition.getVisibleVersion();
+            long partitionId = partition.getId();
+            if (basePartitionInfo.getId() != partitionId || basePartitionInfo.getVersion() < visibleVersion) {
+                needRefreshPartitionNames.addAll(materializedView.getTableMvPartitionNameRefMap().get(basePartitionName));
+                basePartitionVisionMap.put(basePartitionName,
```

Comment:
should update this version map after fresh successfully, not here

Reply:
I think this PR mainly uses insert overwrite, by the way to fix other problems, can we use a new PR for this change?